### PR TITLE
MC-37657: A saved VAT Number is not displayed when placing an order by Admin on a Customer behalf

### LINKED ISSUE #31846: A saved VAT Number is not displayed when placing an order by Admin on a Customer behalf
**Preconditions:**
* Magento EE 2.4.0, EE 2.4.1-beta1 is installed
* A Customer without Billing and Shipping addresses is created
* Go to ther Magento Admin panel > CUSTOMERS > All Customers
* Select the created Customer
* Click the “Edit” link
* Click the “Addresses” vertical tab
* Click the “Add New Address” button
 Default Billing Address = Yes
 Default Shipping Address = Yes
 Street Address = 24 Grosvenor Square
 Country = United Kingdom
 City = London
 ZIP = W1A 1AE
 Phone Number = 1234567
 **VAT Number 584451913**
* Click the “Save” button
* Go to the Storefront as the created Customer > My Account > Address Book
* Make sure that the VAT Number value is displayed

*(see the attached file EE 2.4.1-beta1.VAT ADRESSES STOREFRONT.png)*

**Steps to Reproduce:**
* Go to Magento Admin panel again > SALES > Orders
* Click the “Create New Order” button
* Select the created Customer
* Scroll down the page to the “VAT Number” input field of the “Address Information” section

**(/) Expected results:**

“VAT Number” input fields of both Billing Address and Shipping Address should be filled with a previously entered value (584451913)

**(x) Actual results:**

“VAT Number” input fields of both Billing Address and Shipping Address are empty

*(see the attached EE 2.4.1-beta1.VAT ADMIN ORDER.png)*

### PR DESCRIPTION
**Pull request by engcom-Echo**
### Fixed Issues (if relevant)

1. Fixes magento/magento2#31846

### Manual testing scenarios (*)

1. See: magento/magento2#31846

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
@magento run WebAPI Tests

File: app/code/Magento/Customer/Test/Mftf/Data/CustomerData.xml [hunk 1]
```diff
@@ -233,6 +233,17 @@
         <data key="website_id">0</data>
         <requiredEntity type="address">UK_Not_Default_Address</requiredEntity>
     </entity>
+    <entity name="Customer_With_Vat_Number" type="customer">
```
Comment (zakdma) [inline app/code/Magento/Customer/Test/Mftf/Data/CustomerData.xml]: Could you extend one of the existing datasets and just change/add your address with VAT?
For example: Simple_US_Customer

File: app/code/Magento/Sales/Test/Mftf/ActionGroup/AssertAdminFieldsFilledDefaultBillingAddressActionGroup.xml [hunk 2]
```diff
@@ -0,0 +1,71 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+<actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+              xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+    <actionGroup name="AssertAdminFieldsFilledDefaultBillingAddressActionGroup">
```
Comment (zakdma) [inline app/code/Magento/Sales/Test/Mftf/ActionGroup/AssertAdminFieldsFilledDefaultBillingAddressActionGroup.xml]: Please rename to
AssertAdminBillingAddressFieldsOnOrderCreateFormActionGroup

File: app/code/Magento/Sales/Test/Mftf/ActionGroup/AssertAdminFieldsFilledDefaultBillingAddressActionGroup.xml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        <arguments>
+            <argument name="firstName" type="string" defaultValue="John"/>
+            <argument name="lastName" type="string" defaultValue="Doe"/>
+            <argument name="company" type="string" defaultValue="Magento"/>
+            <argument name="streetLine1" type="string" defaultValue="7700 West Parmer Lane"/>
+            <argument name="streetLine2" type="string" defaultValue="113"/>
+            <argument name="city" type="string" defaultValue="Los Angeles"/>
+            <argument name="postcode" type="string" defaultValue="90001"/>
+            <argument name="phone" type="string" defaultValue="512-345-6789"/>
+            <argument name="vatNumber" type="string" defaultValue="U1234567891"/>
```
Comment (zakdma) [inline app/code/Magento/Sales/Test/Mftf/ActionGroup/AssertAdminFieldsFilledDefaultBillingAddressActionGroup.xml]: Please add all address fields like country, region and others
Check both possibilities for region (dropdown an plain text field)

File: app/code/Magento/Sales/Test/Mftf/Test/AdminCheckingFieldsFilledFromDefaultBillingAddressCustomerInNewOrderTest.xml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+<tests xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/testSchema.xsd">
+    <test name="AdminCheckingFieldsFilledFromDefaultBillingAddressCustomerInNewOrderTest">
+        <annotations>
+            <features value="Sales"/>
+            <stories value="Create order in Admin"/>
+            <title value="Checking fields filled from default billing address customer"/>
+            <description value="Checking fields filled from default billing address customer on create new order page"/>
+            <severity value="AVERAGE"/>
+            <testCaseId value="MC-40646"/>
```
Comment (zakdma) [inline app/code/Magento/Sales/Test/Mftf/Test/AdminCheckingFieldsFilledFromDefaultBillingAddressCustomerInNewOrderTest.xml:17]: Add zephyr test to appropriate test cycle

File: app/code/Magento/Sales/Test/Mftf/Test/AdminCheckingFieldsFilledFromDefaultBillingAddressCustomerInNewOrderTest.xml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+            <useCaseId value="MC-37657"/>
+            <group value="Sales"/>
```
Comment (zakdma) [inline app/code/Magento/Sales/Test/Mftf/Test/AdminCheckingFieldsFilledFromDefaultBillingAddressCustomerInNewOrderTest.xml]: group name should be camelCase
add customer group

**Review by zakdma — CHANGES_REQUESTED** [review]

**Comment by engcom-Echo** [comment]
@magento run all tests

File: app/code/Magento/Sales/Test/Mftf/Test/AdminCheckingFieldsFilledFromDefaultBillingAddressCustomerInNewOrderTest.xml [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+            <group value="sales"/>
+            <group value="customer"/>
+        </annotations>
+        <before>
+            <createData entity="Customer_With_Vat_Number" stepKey="createCustomer"/>
+            <actionGroup ref="AdminLoginActionGroup" stepKey="loginAsAdmin"/>
+        </before>
+        <after>
+            <actionGroup ref="AdminLogoutActionGroup" stepKey="logoutOfAdmin"/>
+            <deleteData createDataKey="createCustomer" stepKey="deleteCustomer"/>
```
Comment (zakdma) [inline app/code/Magento/Sales/Test/Mftf/Test/AdminCheckingFieldsFilledFromDefaultBillingAddressCustomerInNewOrderTest.xml:28]: Data operations should be before any UI operations
In this case move logoutOfAdmin at the bottom of **after** section

**Review by zakdma — APPROVED** [review]

**Comment by zakdma** [comment]
@magento import code to https://github.com/magento-tsg/magento2ce

**Comment by magento-engcom-team** [comment]
@zakdma the branch with code successfully imported into`magento-tsg/magento2ce` repository. Branch name: `imported-magento-magento2-31850`.

**Pull request merged**